src/components/CharacterButton.vue

```diff
@@ -0,0 +1,358 @@
+<template>
+  <q-btn
+    flat
+    class="q-pa-none character-button"
+    :disable="uiLocked"
+    :class="{ opaque: loading }"
+  >
+    <!-- q-imgだとdisableのタイミングで点滅する -->
+    <img
+      v-if="selectedStyleInfo"
+      class="q-pa-none q-ma-none"
+      :src="selectedStyleInfo.iconPath"
+    />
+    <div v-else class="dummy-image"></div>
+    <div v-if="loading" class="loading">
+      <q-spinner color="primary" size="1.6rem" :thickness="7" />
+    </div>
+    <q-menu
+      class="character-menu"
+      transition-show="none"
+      transition-hide="none"
+    >
+      <q-list>
+        <q-item
+          v-for="(characterInfo, characterIndex) in characterInfos"
+          :key="characterIndex"
+          class="q-pa-none"
+        >
+          <q-btn-group flat class="col full-width">
+            <q-btn
+              flat
+              no-caps
+              v-close-popup
+              class="col-grow"
+              :class="
+                characterInfo &&
+                selectedCharacter &&
+                characterInfo.metas.speakerUuid ===
+                  selectedCharacter.metas.speakerUuid &&
+                'selected-character-item'
+              "
+              @click="
+                characterInfo
+                  ? onSelectSpeaker(characterInfo.metas.speakerUuid)
+                  : $emit('update:selectedStyle', undefined)
+              "
```

Comment:
たぶん、キャラクターが居ないときに`$emit('update:selectedStyle', undefined)`が呼ばれる想定だと思うのですが、`characterInfos`が空配列なのでそもそもこのボタンが生成されていない気がします。
ここは↓にしておいて、モーフィングUIのときに修正するのが良いかも？
```suggestion
              @click="onSelectSpeaker(characterInfo.metas.speakerUuid)"
```